### DISCUSSION TITLE
Optional Gamma/Exponential Prior on Flexibility and Asperity Parameters of CF Model

### DISCUSSION THREAD

**Discussion by fmkroci** [NIFTy-PPL/NIFTy#28] [Ideas]
Hello everybody,

First of all thank you very much for developing NIFTy!

I thought of some small (optional) extensions to the correlated field maker model and would be interested to hear your opinion.

I think it might be useful to extend the correlated field model (add_fluctuations specifically) to support other prior distributions rather than just Lognormal.

I encountered situations in which the signals are quite smooth but occasionally need to describe more peaky structures. The exponential would provide a more constrained option while still allowing for some larger values if required.

If you are interested, I would be happy to create a pull request.

[Just for information I only worked with the "classic" NIFTy implementation so far, not with the Jax version]

**Comment by ph-frank** [comment]
Hi,
Thank you for opening this discussion! It's a great proposed extension in the "classic" (old) NIFTy as there are generally cases (e.g. the one you mentioned but also others) where the default prior models on the spectral parameters are sub-optimal.

That being said, `nifty.re`, the jax based version of nifty, already implements this feature. There users have the option to pass arbitrary generative models for the spectral parameters instead of using the default priors. Now I don't want to discourage any contributions to the old nifty, but given that development has moved to `nifty.re` for quite some time now and given that this feature already exists there, it might be difficult to find users for this specific feature.

I see two viable options to move forward here:
- If you want to go ahead and add this to nifty it is a more then welcome contribution! In this case aim to have the feature aligned with the capabilities of the `nifty.re` version of the feature
- Leave the state of nifty as is and focus on updating the ift-pwa work to the latest version of nifty to make use of this (and potentially other) features.

Hope this helps in providing some further insights!

**Reply by fmkroci** [reply]
Hi, 

Thanks a lot for the suggestions. Moving iftpwa  to nifty.re is planned but will be a larger project.

I will check how easily the functionality of nifty.re can be implemented in the old code.
Should I just feel free to open a pull request?

Otherwise, we can work with a custom add_fluctuations without any issues for now. 

Cheers

**Reply by fmkroci** [reply]
@ph-frank Maybe a small follow-up question. Do you already have an implementation of an Exponential distribution somewhere?
At the moment I map normal -> uniform -> exponential but this might be numerically problematic at the edges.

**Reply by fmkroci** [reply]
> @ph-frank Maybe a small follow-up question. Do you already have an implementation of an Exponential distribution somewhere? At the moment I map normal -> uniform -> exponential but this might be numerically problematic at the edges.

I just found that jax supports scipy.stats.norm.logcdf

**Reply by ph-frank** [reply]
Afaik we don't have the exponential prior implemented yet. Happy to have it as an addition to nifty! Feel free to open pull requests for these features at any time, we can get the numerics sorted out there at a later stage :)
